main.cpp

```diff
@@ -39,29 +39,427 @@ Purpose:  This project continues developing Project3.
+{
+    std::cout << "Compressor destructed." << "\n";
+}
+
+Compressor::Saturator::Saturator()
+{
+    std::cout << "Nested type Saturator constructed." << "\n";
+}
+
+Compressor::Saturator::~Saturator()
+{
+    std::cout << "Nested type Saturator destructed." << "\n";
+}
+
+void Compressor::compress(double audioIn)
+{
+    if (audioIn > threshold)
+    {
+        if (compType == 0)
+            std::cout << "Compressing with ratio of: " << ratio << " using Peak algorithm." << std::endl << "Threshold is set to: " << threshold << std::endl << "Output gain is set to: " << outputGain << std::endl;
+        else
+            std::cout << "Compressing with ratio of: " << ratio << " using RMS algorithm." << std::endl << "Threshold is set to: " << threshold << std::endl << "Output gain is set to: " << outputGain << std::endl;
+    }
+}
+
+void Compressor::volumeMakeUp()
+{
+    float difference = inputGain - outputGain;
+    std::cout << "Adjusting the volume by " << difference << std::endl;
+}
+
+void Compressor::saturateInput(Saturator sat)
+{
+    std::cout << "Saturation engaged! Saturator type: " << sat.satType << std::endl << "Mix parameter is set to: " << sat.mix << std::endl;
+}
+
+double Compressor::rampThreshold(double desiredThreshold)
+{
+    if (desiredThreshold < threshold)
+    {
+        while (threshold >= desiredThreshold)
+        {
+            std::cout << "Desired threshold is smaller, ramping down threshold: " << threshold << "\n"; 
+            threshold -= 0.1;
+
+            if (threshold <= desiredThreshold)
+            {
+                std::cout << "Procedure completed.\n";
+                return threshold;
+            }
+        }
+    }
+    else
+    {
+        while (threshold <= desiredThreshold)
+        {
+            std::cout << "Desired threshold is bigger, ramping up threshold: " << threshold << "\n";
+            threshold += 0.1;
+
+            if (threshold >= desiredThreshold)
+            {
+                std::cout << "Procedure completed.\n";
+                return threshold;   
+            }
+        }
+    }
+    std::cout << "Desired threshold is the same as initial threshold." << "\n";
+    return threshold;
+}
+
 /*
  copied UDT 2:
  */
 
+struct Bakery
+{
+    Bakery();
+    ~Bakery();
+    double flourAmount;
+    int numCake;
+    int numBread;
+    float numMoney;
+    int maxOvenTemp;
+    int minutesLeft;
+    std::string name;
+
+    struct RyeBread
+    {
+        RyeBread();
+        ~RyeBread();
+        int bakingTime = 45;
+        float bakingTemp = 350.0f;
+        float numPumpkinSeeds = 5.5f;
+        float numLinseed = 2.3f;
+        bool bioIngredients = 1;
+
+        void mixIngredients();
+        float getWeight(float flourUsed, float numPumpkinSeeds, float numLinseed);
+        bool burn(int bakingTime);
+    };
+
+    void bakeBread(RyeBread brd);
+    float cakeSell(std::string cakeName);
+    void smellGreat();
+    int minuteTimer(int timeAmount);
```

Comment:
These function names don't really contain actions. 
`sellCake` and `createTimer` are better names.  the `verb` is always first in the name of the function.
- [ ] revise